### PR TITLE
Fix recompile in torchax pt runner

### PR DESCRIPTION
**Pull request by lsy323**
# Description

If a jitted module is called with args placed onto device with `jax.device_put`, the HLO graph would contain replicated sharding annotation on the input arg if it's placed on a single TPU device.

By looking into the HLO graph dumps, I found during serving, all the HLO graphs contain the `replicated` sharding annotation. So during warmup time, we use `device_put` to enforce the sharding annotation on the graph inputs. Therefore we can avoid recompilation during serving.

Some minor clean up.

# Tests

Ran benchmarking, compare number of XLA graph dumps after server warm up and after a benchmarking run, the XLA graph dump size is the same before and after.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [ x ] I have performed a self-review of my code.
- [ x ] I have necessary comments in my code, particularly in hard-to-understand areas.
- [ x ] I have made or will make corresponding changes to any relevant documentation.

### CODE REVIEW

**Comment by lsy323** [comment]
I'll just merge the change, since torchax + PT runner is meant to be experimental (Analyzing torchax model wrapper purpose), and it doesn't affect other code path. Merge the change so that we can track the torchax performance on the dashboard
